### ISSUE TITLE
implementing a ConstantKernel

### ISSUE THREAD

**Issue by andreaskoher** [JuliaGaussianProcesses/TemporalGPs.jl#56]
Hi @willtebbutt,

I am very impressed by the speed improvement of TemporalGPs over the standard approach and it would be great to see a few more kernels. To begin with, I took the constant kernel as the simplest possible extension and following your implementation and  the notes on [page 263](https://users.aalto.fi/~asolin/sde-book/sde-book.pdf), I added the following custom functions:

```julia
function TemporalGPs.to_sde(k::ConstantKernel{T}, ::SArrayStorage{T}) where {T<:Real}
    F = SMatrix{1, 1, T}(0)
    q = convert(T, 0)
    H = SVector{1, T}(1)
    return F, q, H
end

function TemporalGPs.stationary_distribution(k::ConstantKernel{T}, ::SArrayStorage{T}) where {T<:Real}
    #t0 = only(k.c)
    return TemporalGPs.Gaussian(
        SVector{1, T}(0),
        SMatrix{1, 1, T}(1),
    )
end
```
Now, I can sample from a GP with a constant kernel but suspiciously the posterior is way off:

```julia
import Random
Random.seed!(1)
f_naive = GP(ConstantKernel())
f = to_sde(f_naive, SArrayStorage(Float64))
x = RegularSpacing(0.0, 0.1, 1000)
fx = f(x, .001)
y = rand(fx)

f_post = posterior(fx, y)
y_post = rand(f_post(x))

plot(y)
plot!(y_post)
```
![prior_vs_posterior](https://user-images.githubusercontent.com/18704101/113757019-6a168b80-9712-11eb-81ed-498395f7bdd9.png)

Probably, I have missed something in your implementation and it would be great if you could hint at the problem.

Many thanks!

**Comment by willtebbutt** [comment]
Firstly, thanks for the being the first person to open an issue, and to attempt to implement a new kernel 🎉 I completely agree that the package needs better coverage of what's out there, so it's very welcome.

It looks to me like you've found a bug of some kind in my implementation, which is excellent. It looks like there's something fishy going on with my conditioning algorithm for the posterior over the last element in a sequence. Specifically, I can produce something similar with a `Matern52Kernel`. I think it's more obvious for `ConstantKernel` because it forgets about the distribution over the last state very slowly.

Specifically, something like
```julia
import Random
Random.seed!(1)
f_naive = GP(Matern52Kernel())
f = to_sde(f_naive, SArrayStorage(Float64))
x = RegularSpacing(0.0, 0.1, 50)
fx = f(x, .001)
y = rand(fx)

f_post = posterior(fx, y)
y_post = rand(f_post(x))

plot(y)
plot!(y_post)
```
![Screenshot 2021-04-06 at 19 39 05](https://user-images.githubusercontent.com/3628294/113761823-d2b03900-970f-11eb-9670-a7beaba7202e.png)

I'm going to attempt to debug now.

**Comment by willtebbutt** [comment]
I'm also rather concerned that this managed to get past my tests 😂

**Comment by willtebbutt** [comment]
Looks to be an issue with `rand`. The marginals look fine:
![Screenshot 2021-04-06 at 19 47 13](https://user-images.githubusercontent.com/3628294/113762872-f32cc300-9710-11eb-816c-b998c5bfba28.png)
This would explain how it got by my tests -- the tests for random number generation aren't as thorough as those for things that are deterministic... I guess this is the result haha.

**Comment by willtebbutt** [comment]
@andreaskoher once #57 is merged, could I tempt you to open a PR to contribute this kernel implementation?

**Comment by andreaskoher** [comment]
Sure, I am happy to contribute.

**Comment by willtebbutt** [comment]
Closing because the bug was resolved by #57 and the kernel implementation was added in #58.
